### PR TITLE
Add recursemode parameter to apply ACLs recursively

### PR DESCRIPTION
**Pull request by dobbymoodge**
Same changes as dobbymoodge/puppet-acl#12 but with additional fixes

### CODE REVIEW

File: lib/puppet/type/acl.rb [hunk 1]
```diff
@@ -184,7 +195,7 @@ def insync?(is)
         s = p.tr '-', ''
         r << (s.sub!('r', '')?'r':'-')
         r << (s.sub!('w', '')?'w':'-')
-        r << (s.sub!('x', '')?'x':'-')
+        r << (s.sub!('x', '')?'x':'-')#'
```
Comment (roidelapluie) [inline lib/puppet/type/acl.rb]: are you sure about this line?

Comment (dobbymoodge) [inline lib/puppet/type/acl.rb]: It's a hack to fix emacs' broken ruby parsing. I hate it, but there it is. :(

Comment (dobbymoodge) [inline lib/puppet/type/acl.rb]: i.e. I should fix `ruby-mode` at some point, I guess, but then I'd have to learn ruby AND elisp. :D

Comment (roidelapluie) [inline lib/puppet/type/acl.rb]: github syntay colorinq is also broken it seems

Comment (dobbymoodge) [inline lib/puppet/type/acl.rb]: Does `?'` have special meaning in ruby? That seems to be where github and emacs are freaking out, in L196

Comment (dobbymoodge) [inline lib/puppet/type/acl.rb]: Looks like the ternary operator needs spaces to be digestible to other syntax highlighters as well: 
bad_ternary_parsing

Comment (roidelapluie) [inline lib/puppet/type/acl.rb]: Looks like http://stackoverflow.com/a/1878406

Comment (dobbymoodge) [inline lib/puppet/type/acl.rb]: @roidelapluie https://github.com/dobbymoodge/puppet-acl/pull/14

Comment (dobbymoodge) [inline lib/puppet/type/acl.rb]: Things that can't be parsed unambiguously by either the ruby interpreter or by various syntax highlighters:
```ruby
(false|true?'a':'b')
```
`SyntaxError: unexpected tSTRING_BEG, expecting keyword_do or '{' or '('
(false|true?'a':'b')`

```ruby
(false|true ?'a':'b')
```
Works in the interpreter, but at least `pry` gets it wrong syntax highlighting (github too! Look at the closing paren)
```ruby
((false|true)?'a':'b')
```
Interpreter parses correctly as expected, but `?'a':` still seems to confuse some highlighters.

*sigh*

**Pull request merged**